datarepo-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQueryJade.java

```diff
@@ -0,0 +1,240 @@
+    }
+    projectId = parameters.get(0);
+    datasetName = parameters.get(1);
+  }
+
+  private static String testRunTableName = "testRun";
+  private static String testScriptResultsTableName = "testScriptResults";
+  private static String measurementCollectionTableName = "measurementCollection";
+
+  /**
+   * Upload the test results saved to the given directory. Results may include Test Runner
+   * client-side output and any relevant measurements collected.
+   */
+  public void uploadResults(Path outputDirectory) throws Exception {
+    // get a BigQuery client object
+    logger.debug("BigQuery project_id:dataset_name: {}:{}", projectId, datasetName);
+    BigQuery bigQueryClient = BigQueryUtils.getClient(projectId);
+
+    // read in TestConfiguration TestRunSummary, and array of
+    // MeasurementCollectionScript.MeasurementResultSummary objects
+    renderedTestConfiguration = TestRunner.getRenderedTestConfiguration(outputDirectory);
+    testRunSummary = TestRunner.getTestRunSummary(outputDirectory);
+    measurementCollectionSummaries =
+        MeasurementCollector.getMeasurementCollectionSummaries(outputDirectory);
+
+    // insert a single row into testRun
+    if (BigQueryUtils.checkRowExists(
+        bigQueryClient, projectId, datasetName, testRunTableName, "id", testRunSummary.id)) {
+      logger.info(
+          "A row with this id already exists in the "
+              + testRunTableName
+              + " table. Inserting a duplicate.");
+    }
+    TableId tableId = TableId.of(datasetName, testRunTableName);
+    InsertAllResponse response =
+        bigQueryClient.insertAll(
+            InsertAllRequest.newBuilder(tableId).addRow(buildTestRunRow(outputDirectory)).build());
+    if (response.hasErrors()) {
+      logger.info("hasErrors is true");
+      // If any of the insertions failed, this lets you inspect the errors
+      for (Map.Entry<Long, List<BigQueryError>> entry : response.getInsertErrors().entrySet()) {
+        entry.getValue().forEach(bqe -> logger.info("bqerror: {}", bqe.toString()));
+      }
+    }
+    logger.info("Test run summary written to: {}.{}.{}", projectId, datasetName, testRunTableName);
+
+    // insert into testScriptResults
+    tableId = TableId.of(datasetName, testScriptResultsTableName);
+    InsertAllRequest.Builder request = InsertAllRequest.newBuilder(tableId);
+    for (int ctr = 0; ctr < renderedTestConfiguration.testScripts.size(); ctr++) {
+      TestScriptSpecification testScriptSpecification =
+          renderedTestConfiguration.testScripts.get(ctr);
+      TestScriptResult.TestScriptResultSummary testScriptResult =
+          testRunSummary.testScriptResultSummaries.get(ctr);
+      request.addRow(buildTestScriptResultsRow(testScriptSpecification, testScriptResult));
+    }
+    response = bigQueryClient.insertAll(request.build());
+    if (response.hasErrors()) {
+      logger.info("hasErrors is true");
```

Comment:
Just thinking about the debugging process: It might be helpful to add a little more context to this message, in particular to differentiate it from the log on L89.

Reply:
Good idea, will do.